File: two_sum_ii/two_sum_ii.java
public class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer, Integer> hmap = new HashMap<Integer, Integer>();
        int num_len = numbers.length;
        int i = 0;
        while (i < num_len){
        	int number = numbers[i];
        	
        	if (number == target / 2 && hmap.get(number) != null){
        	    return new int[] {hmap.get(number) + 1,i + 1};
        	}
        	hmap.put(number,i);
        	if (hmap.get(target-number) != null && hmap.get(target-number) != i){
        	        return new int[] {hmap.get(target-number) + 1,i + 1};
        	}
        	i++;
        }
        int[] arr = {0,2}; 
        return arr;
    }
}

File: two_sum_ii/two_sum_ii.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
	num_len = nums.length
	dic = {}
	i = 0 
  while i < num_len	
  	number = nums[i]
  	if number == target / 2 && dic[number]
  		return [dic[number] + 1,i + 1]
  	end
  	dic[number] = i
  	maybe = dic[target-number]
  	if maybe && maybe != i
  		return [maybe + 1,i + 1]
  		break
  	end
    i += 1
  end
end